crates/node/src/validate.rs

```diff
@@ -110,6 +154,33 @@ pub async fn validate(
     conn_pool: &ConnectionPool,
     contract_registry: &ContentAddress,
     block: &Block,
+) -> Result<ValidateOutcome, ValidationError> {
+    let dry_run = DryRun::new(conn_pool.clone(), block).await?;
+    let db_type = Db::DryRun(dry_run);
+    validate_inner(db_type, contract_registry, block).await
+}
+
+/// Validates a block.
+///
+/// Returns a `ValidationResult` if no `ValidationError` occurred that prevented the block from being validated.
+#[cfg_attr(feature = "tracing", tracing::instrument(skip_all))]
+pub(crate) async fn validate_incoming_block(
```

Comment:
Maybe `validate_in_db` or similar might be a better name for this? Just wondering if `validate_incoming_block` might still be a bit ambiguous. Or maybe we can mention in the docs that, despite passing the `Block` as an argument, we actually query its state from the DB?

Reply:
Changed this to `validate`.